src/System.Net.Primitives/src/System/Net/CookieContainer.cs

```diff
@@ -0,0 +1,1062 @@
+                        }
+
+                        if (to_add)
+                        {
+                            // In 'source' are already orederd.
+                            // If two same cookies come from different 'source' then they
+                            // will follow (not replace) each other.
+                            destination.InternalAdd(cookie, false);
+                        }
+                    }
+                }
+            }
+        }
+
+        public string GetCookieHeader(Uri uri)
+        {
+            if (uri == null)
+            {
+                throw new ArgumentNullException("uri");
+            }
+            string dummy;
+            return GetCookieHeader(uri, out dummy);
+        }
+
+        internal string GetCookieHeader(Uri uri, out string optCookie2)
+        {
+            CookieCollection cookies = InternalGetCookies(uri);
+            string cookieString = String.Empty;
+            string delimiter = String.Empty;
+
+            foreach (Cookie cookie in cookies)
+            {
+                cookieString += delimiter + cookie.ToString();
+                delimiter = "; ";
+            }
+            optCookie2 = cookies.IsOtherVersionSeen ?
+                          (Cookie.SpecialAttributeLiteral +
+                           Cookie.VersionAttributeName +
+                           Cookie.EqualsLiteral +
+                           Cookie.MaxSupportedVersion.ToString(NumberFormatInfo.InvariantInfo)) : String.Empty;
+
+            return cookieString;
+        }
+
+        public void SetCookies(Uri uri, string cookieHeader)
+        {
+            if (uri == null)
+            {
+                throw new ArgumentNullException("uri");
+            }
+            if (cookieHeader == null)
+            {
+                throw new ArgumentNullException("cookieHeader");
+            }
+            CookieCutter(uri, null, cookieHeader, true); // Will throw on error
+        }
+    }
+
+    internal class PathList
+    {
+        private SortedList _list = (SortedList.Synchronized(new SortedList(PathListComparer.StaticInstance)));
+
+        public PathList()
+        {
+        }
+
+        public int Count
+        {
+            get
+            {
+                return _list.Count;
+            }
+        }
+
+        public int GetCookiesCount()
+        {
+            int count = 0;
+            lock (SyncRoot)
+            {
+                foreach (CookieCollection cc in _list.Values)
+                {
+                    count += cc.Count;
+                }
+            }
+            return count;
+        }
+
+        public ICollection Values
+        {
+            get
+            {
+                return _list.Values;
+            }
+        }
+
+        public object this[string s]
+        {
+            get
+            {
+                return _list[s];
+            }
+            set
+            {
+                lock (SyncRoot)
```

Comment:
You're correct, but I think leaving the locking in for now will make the eventual move to SortedList<> more straightforward.